ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPoolTest.java

```diff
@@ -471,18 +472,56 @@ public void getAttestationsForBlock_shouldNotAddAttestationsFromWrongFork() {
 
   @TestTemplate
   public void getAttestations_shouldReturnAllAttestations() {
-    // TODO EIP7549 Handle Electra attestations
+    final AttestationData firstAttestationData = dataStructureUtil.randomAttestationData();
+    final Attestation firstAttestation =
+        addAttestationFromValidators(firstAttestationData, 1, 2, 3);
+    final AttestationData secondAttestationData = dataStructureUtil.randomAttestationData();
+    final Attestation secondAttestation =
+        addAttestationFromValidators(secondAttestationData, 3, 4, 5);
+    assertThat(aggregatingPool.getAttestations(Optional.empty(), Optional.empty()))
+        .containsExactlyInAnyOrder(firstAttestation, secondAttestation);
+  }
+
+  @TestTemplate
+  public void getAttestations_shouldReturnElectraAttestationsOnly_WhenAtLeastOneIsPresent() {
+    // Genesis spec must be before Electra in order to be able to add phase0 attestations
     assumeThat(specMilestone).isLessThan(ELECTRA);
-    final AttestationData attestationData = dataStructureUtil.randomAttestationData();
-    Attestation attestation = addAttestationFromValidators(attestationData, 1, 2, 3);
+    final Spec mockedSpec = mock(Spec.class);
+    final AggregatingAttestationPool aggregatingPool =
+        new AggregatingAttestationPool(
+            mockedSpec,
+            mockRecentChainData,
+            new NoOpMetricsSystem(),
+            DEFAULT_MAXIMUM_ATTESTATION_COUNT);
+    // Adding a phase0 attestation to the aggregation pool
+    final Spec phase0Spec = TestSpecFactory.createMinimalPhase0();
+    when(mockedSpec.atSlot(any())).thenReturn(phase0Spec.getGenesisSpec());
+    final DataStructureUtil phase0DataStructureUtil = new DataStructureUtil(phase0Spec);
+    final AttestationData phase0AttestationData =
+        phase0DataStructureUtil.randomAttestationData(SLOT.minus(4));
+    addAttestationFromValidators(aggregatingPool, phase0AttestationData, phase0Spec, 4, 5);
```

Comment:
this is interesting - beacon-api kind of implied in a new issue that the better behaviour woudl be to convert any non electra attestations to electra and return them too...

Reply:
Well, we agreed to keep filtering Electra attestations only. The issue is closed